plugins/woocommerce/src/Internal/Traits/RestApiCache.php

```diff
@@ -741,7 +780,10 @@ protected function get_key_info_for_cached_response( WP_REST_Request $request, b
 		);
 
 		if ( $vary_by_user ) {
-			$user_id           = wc_get_container()->get( LegacyProxy::class )->call_function( 'get_current_user_id' );
+			/** @var LegacyProxy $legacy_proxy Container returns mixed, cast for type safety. */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+			$legacy_proxy = wc_get_container()->get( LegacyProxy::class );
+			// @phpstan-ignore-next-line argument.type -- get_current_user_id returns int at runtime.
+			$user_id           = intval( $legacy_proxy->call_function( 'get_current_user_id' ) );
 			$cache_key_parts[] = "user_{$user_id}";
```

Comment:
Ideally, instead of ignoring the error the `@var` could be used to set the type
```suggestion
			/** @var int $user_id */
			$user_id           = $legacy_proxy->call_function( 'get_current_user_id' );
			$cache_key_parts[] = 'user_' . intval( $user_id );
```